OpenSim/Moco/MocoGoal/MocoStepLengthAsymmetryGoal.h

```diff
@@ -0,0 +1,162 @@
+#ifndef OPENSIM_MOCOSTEPLENGTHASYMMETRY_H
+#define OPENSIM_MOCOSTEPLENGTHASYMMETRY_H
+/* -------------------------------------------------------------------------- *
+ * OpenSim: MocoStepLengthAsymmetryGoal.h                                     *
+ * -------------------------------------------------------------------------- *
+ * Copyright (c) 2020 Stanford University and the Authors                     *
+ *                                                                            *
+ * Author(s): Russell Johnson, Nicholas Bianco                                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0          *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "MocoGoal.h"
+
+namespace OpenSim {
+
+/** Minimize the error between a model's step time asymmetry and a specified
+target asymmetry value over a gait cycle.
+
+Step Length Asymmetry (SLA) is a percentage and is calculated as follows:
+The Right Step Length (RSL) is the distance between feet at right foot strike
+The Left Step Length (LSL) is the distance between feet at left foot strike
+Step Length Asymmetry = (RSL - LSL) / (RSL + LSL)
+
+In this goal, the distance between feet throughout the gait cycle is controlled
```

Comment:
update the comments here to mirror the comments in the `example2DWalking_StepAsymmetry.m` script for step length asymmetry.